Add sidecar containers

MartinWeise opened the pull request:
Closes #260 

This PR adds sidecar support.

We need this sidecar in our database repository [1] project to import and export .csv files from minIO. We wrote an image for this and would like to continue using the mariadb-operator, we are very happy with it. We believe that extending the CRD by adding sidecars, other projects can extend their implementation too.

This is my first-ever PR on Github, I hope I did not miss anything.

[1] https://www.ifs.tuwien.ac.at/infrastructures/dbrepo/latest/

mmontes11 commented on pkg/builder/statefulset_container_builder.go:
```diff
@@ -33,6 +33,19 @@ func (b *Builder) buildStsContainers(mariadb *mariadbv1alpha1.MariaDB, dsn *core
+	if mariadb.Spec.SidecarContainers != nil {
+		for index, containerTpl := range mariadb.Spec.InitContainers {
+			sidecarContainer := buildContainer(&containerTpl)
+			sidecarContainer.Name = fmt.Sprintf("init-%d", index)
```
```suggestion
			sidecarContainer.Name = fmt.Sprintf("sidecar-%d", index)
```

mmontes11 replied:
This is to avoid clashing with the `initContainers`

MartinWeise replied:
Agree, I overlooked it, thank you.

mmontes11 commented on examples/manifests/mariadb_v1alpha1_mariadb_sidecar.yaml:
```diff
@@ -0,0 +1,31 @@
+    - image: busybox:1.36
+      args:
+        - sleep
+        - "9999999"
```
```suggestion
        - infinity
```

mmontes11 commented on pkg/builder/statefulset_container_builder.go:
```diff
@@ -33,6 +33,19 @@ func (b *Builder) buildStsContainers(mariadb *mariadbv1alpha1.MariaDB, dsn *core
 		containers = append(containers, buildMetricsContainer(mariadb.Spec.Metrics, dsn))
 	}
+	if mariadb.Spec.SidecarContainers != nil {
+		for index, containerTpl := range mariadb.Spec.InitContainers {
```
```suggestion
		for index, containerTpl := range mariadb.Spec.SidecarContainers {
```

mmontes11 replied:
Probably you've meant `SidecarContainers` here right?

mmontes11 reviewed (changes requested):
Hey! thanks for your contribution. It's looking good but some adjustments are needed, could you please take a look?

Also, make sure you run `make gen` and commit the changes to re-generate the CRDs

MartinWeise commented:
Thank you for your review, I think I addressed the points and accepted the change requests, also I re-generated the CRDs as pointed out, I must have overlooked that part in the contributors guide. I am finished with my changes, if you could review again?

MartinWeise reviewed (commented):
Thank you for the meaningful review, I finished my commits, please verify again!

MartinWeise reviewed (commented):
All checks have passed, I think we are good to go :-)

mmontes11 reviewed (approved):
LGTM, thanks for your contribution 🙏🏻

MartinWeise commented:
Thank you so much, I appreciate the quick response and professional attitude, it was fun contributing!
